### PR TITLE
Add default argument for PetscParMatrix constructor out of a HyprePar…

### PR DESCRIPTION
**Pull request by stefanozampini**
…Matrix or a general Operator

### CODE REVIEW

File: linalg/petsc.hpp [hunk 1]
```diff
@@ -181,7 +181,7 @@ class PetscParMatrix : public Operator
        PETSc format @a tid. */
    /** The supported type ids are: Operator::PETSC_MATAIJ,
        Operator::PETSC_MATIS, and Operator::PETSC_MATSHELL. */
-   PetscParMatrix(const HypreParMatrix *ha, Operator::Type tid);
+   PetscParMatrix(const HypreParMatrix *ha, Operator::Type tid = Operator::PETSC_MATAIJ);
```
Comment (v-dobrev) [inline linalg/petsc.hpp]: I think we should add `explicit` to this constructor.

File: linalg/petsc.hpp [hunk 2]
```diff
@@ -181,7 +181,8 @@ class PetscParMatrix : public Operator
        PETSc format @a tid. */
    /** The supported type ids are: Operator::PETSC_MATAIJ,
        Operator::PETSC_MATIS, and Operator::PETSC_MATSHELL. */
-   PetscParMatrix(const HypreParMatrix *ha, Operator::Type tid);
+   PetscParMatrix(const HypreParMatrix *ha,
+                  Operator::Type tid = Operator::PETSC_MATAIJ);
```
Comment (v-dobrev) [inline linalg/petsc.hpp]: [Posting my comment again since it was hidden by Tzanio's commit]:
I think we should add `explicit` to this constructor.

Comment (tzanio) [reply linalg/petsc.hpp]: @stefanozampini do you have any objections to label the constructor `explicit`?

**Comment by stefanozampini** [comment]
no

Il 18 Dic 2017 9:17 PM, "Tzanio Kolev" <notifications@github.com> ha
scritto:

**Pull request merged**